File: BMS.WEB/js/modules/master_tables/Functionary.js
Ext.onReady(function() {

    var aspPageFunctionary = 'Functionary.aspx';

    var aspPageCity = 'City.aspx';

    var aspPageHeadquarter = 'Headquarter.aspx';

    var aspPageCharge = 'Charge.aspx';

    function Functionary() {
        this.Id_Functionary = null,
        this.Id_BiometricReader = null,
        this.Id_Company = null,
        this.Id_Headquarter = null,
        this.Name = null,
        this.LastName = null,
        this.BirthdayMonth = null,
        this.BirthdayDay = null,
        this.Address = null,
        this.Id_City = null,
        this.PostalZone = null,
        this.ZipCode = null,
        this.AirSection = null,
		this.TelephoneNumber = null,
        this.CelphoneNumber = null,
        this.BeeperNumber = null,
        this.BeeperCode = null,
        this.FaxNumber = null,
        this.Email = null,
        this.Password = null,
        this.Id_Charge = null,
        this.ByPayroll = null,
        this.ByHour = null,
        this.IntegralSalary = null,
        this.ContractClass = null,
        this.IsActive = null,
        this.Id_Dependency = null,
        this.Id_CostCenter = null,
        this.Turn = null,
        this.ExtensionNumber = null,
        this.BossName = null,
        this.BossExtensionNumber = null,
        this.Observations = null,
        this.IndexReader = null,
        this.Id_Between = null,
        this.FingerPrint = null,
        this.Photo = null,
        this.Id_UserCreateRegistration = null,
        this.DateCreateRegistration = null,
        this.Id_UserModifyRegistration = null,
        this.DateModifyRegistration = null
    };

    var companyStore = new Ext.data.Store({
        fields: [{ name: 'Id_Third' }, { name: 'Name'}],
        data: []
    });

    var companyCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_company_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_Third',
        displayField: 'Name',
        valueField: 'Id_Third',
        queryMode: 'local',
        store: companyStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });

    loadCombo(aspPageFunctionary, 'GetCompany', "{'start':0,'limit':0}", companyStore, companyCombo);


    var headquarterStore = new Ext.data.Store({
        fields: [{ name: 'Id_Headquarter' }, { name: 'Description'}],
        data: []
    });

    var headquarterCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_headquarter_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_Headquarter',
        displayField: 'Description',
        valueField: 'Id_Headquarter',
        queryMode: 'local',
        store: companyStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });

    loadCombo(aspPageHeadquarter, 'List', "{'start':0,'limit':0}", headquarterStore, headquarterCombo);

    var chargeStore = new Ext.data.Store({
        fields: [{ name: 'Id_Charge' }, { name: 'Name'}],
        data: []
    });

    var chargeCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_charge_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_Charge',
        displayField: 'Name',
        valueField: 'Id_Charge',
        queryMode: 'local',
        store: companyStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });

    loadCombo(aspPageCharge, 'List', "{'start':0,'limit':0}", chargeStore, chargeCombo);

    var cityStore = new Ext.data.Store({
        fields: [{ name: 'Id_City' }, { name: 'Name'}],
        data: []
    });

    var cityCombo = Ext.create('Ext.form.field.ComboBox', {
        id: 'id_city_combo',
        mode: 'local',
        triggerAction: 'all',
        forceSelection: true,
        editable: false,
        name: 'Id_City',
        displayField: 'Name',
        valueField: 'Id_City',
        queryMode: 'local',
        store: cityStore,
        listeners: {
            select: function(combo, arrRec, obj) { }
        }
    });

    loadCombo(aspPageCity, 'List', "{'start':0,'limit':0}", cityStore, cityCombo);

    var MasterRowEditorPerson = new Ext.grid.plugin.RowEditing({
        errorSummary: false,
        listeners: {
            validateedit: function(editor, e, eOpts) {
                delete e.newValues[''];
                delete e.newValues['Photo'];

                saveData(
                    aspPageFunctionary,
                    'Save',
                    'objProperties',
                    e.newValues,
                    function(data) {
                        MasterStorePerson.load();
                        //loadData(aspPageFunctionary, 'List', "{'start':0,'limit':20}", MasterGridPerson.getStore(), null, null);
                    },
                    null
                );
            }
        }
    });

    Ext.define('MasterModelPerson', {
        extend: 'Ext.data.Model',
        fields: [
			{ name: 'Id_Functionary' },
            { name: 'Id_BiometricReader' },
            { name: 'Id_Company' },
            { name: 'Id_Headquarter' },
            { name: 'Name' },
            { name: 'LastName' },
            { name: 'BirthdayMonth' },
            { name: 'BirthdayDay' },
            { name: 'Address' },
            { name: 'Id_City' },
            { name: 'PostalZone' },
            { name: 'ZipCode' },
            { name: 'AirSection' },
            { name: 'TelephoneNumber' },
            { name: 'CelphoneNumber' },
            { name: 'BeeperNumber' },
            { name: 'BeeperCode' },
            { name: 'FaxNumber' },
            { name: 'Email' },
            { name: 'Password' },
            { name: 'Id_Charge' },
            { name: 'ByPayroll' },
            { name: 'ByHour' },
            { name: 'IntegralSalary' },
            { name: 'ContractClass' },
            { name: 'IsActive' },
            { name: 'Id_Dependency' },
            { name: 'Id_CostCenter' },
            { name: 'Turn' },
            { name: 'ExtensionNumber' },
            { name: 'BossName' },
            { name: 'BossExtensionNumber' },
            { name: 'Observations' },
            { name: 'IndexReader' },
            { name: 'Id_Between' },
            { name: 'FingerPrint' },
            { name: 'Photo' },
            { name: 'Id_UserCreateRegistration' },
            { name: 'DateCreateRegistration' },
            { name: 'Id_UserModifyRegistration' },
            { name: 'DateModifyRegistration' }
		],
        idProperty: 'Id_Functionary'
    });

    var MasterStorePerson = new Ext.data.Store({
        pageSize: 20,
        model: 'MasterModelPerson',
        remoteSort: true,
        proxy: {
            type: 'jsonp',
            url: aspPageFunctionary,
            reader: {
                root: 'result',
                totalProperty: 'total'
            },
            simpleSortMode: true,
            extraParams: {
                accion: 'List'
            }
        },
        sorters: [{
            property: 'Id_Functionary',
            direction: 'ASC'
        }]
    });

        MasterStorePerson.load();

        var MasterGridPerson = new Ext.grid.GridPanel({
            frame: false,
            border: true,
            width: Ext.getBody().getViewSize().width,
            height: Ext.getBody().getViewSize().height,
            monitorResize: true,
            stripeRows: true,
            columnLines: true,
            stateful: true,
            stateId: 'grid',
            store: MasterStorePerson,
            columns: [
				Ext.create('Ext.grid.RowNumberer'),
				{ header: "Foto", width: 70, dataIndex: 'Photo', renderer: setPhoto },
                { text: 'Documento de Identificaci&oacute;n', width: 160, dataIndex: 'Id_Functionary', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Identificador Lector', width: 150, dataIndex: 'Id_BiometricReader', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Nombre', width: 150, dataIndex: 'Name', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Apellido', width: 150, dataIndex: 'LastName', editor: new Ext.form.TextField({ allowBlank: false }) },
				{ text: 'Cargo', width: 150, dataIndex: 'Id_Charge', renderer: function(val, meta, rec) {
				    return getValueFromStore(val, meta, rec, chargeStore, 'Id_Charge', 'Name');
				}, editor: chargeCombo
				},
				{ text: 'Empresa', width: 150, dataIndex: 'Id_Company', renderer: function(val, meta, rec) {
				    return getValueFromStore(val, meta, rec, companyStore, 'Id_Third', 'Name');
				}, editor: companyCombo
				},
				{ text: 'Sede', width: 150, dataIndex: 'Id_Headquarter', renderer: function(val, meta, rec) {
				    return getValueFromStore(val, meta, rec, headquarterStore, 'Id_Headquarter', 'Description');
				}, editor: headquarterCombo
				},
				{ text: 'Password', width: 80, dataIndex: 'Password', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Mes Nacimiento', dataIndex: 'BirthdayMonth', editor: new Ext.form.TextField({}), renderer: function(val, meta, rec) {
                    return getMonthFromInt(val);
                }
                },
                { text: 'D&iacute;a Nacimiento', dataIndex: 'BirthdayDay', editor: new Ext.form.TextField({}) },
                { text: 'Direcci&oacute;n', dataIndex: 'Address', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Ciudad', dataIndex: 'Id_City', renderer: function(val, meta, rec) {
                    return getValueFromStore(val, meta, rec, cityStore, 'Id_City', 'Name');
                }, editor: cityCombo
                },
				{ text: 'Apartado A&eacute;reo', dataIndex: 'AirSection', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Tel&eacute;fono', dataIndex: 'TelephoneNumber', editor: new Ext.form.TextField({}) },
                { text: 'Celular', dataIndex: 'CelphoneNumber', editor: new Ext.form.TextField({}) },
				{ text: 'N&uacute;mero de Beeper', dataIndex: 'BeeperNumber', editor: new Ext.form.TextField({ allowBlank: false }) },
				{ text: 'C&oacute;digo de Beeper', dataIndex: 'BeeperCode', editor: new Ext.form.TextField({ allowBlank: false }) },
                { text: 'Email', dataIndex: 'Email', editor: new Ext.form.TextField({ allowBlank: false }) },
				{ text: 'Nombre del Jefe', dataIndex: 'BossName', editor: new Ext.form.TextField({ allowBlank: false }) },
				{ text: 'Observaciones', width: 150, dataIndex: 'Observations', editor: new Ext.form.TextField({ allowBlank: false }) },
				{ text: 'Turno?', dataIndex: 'Turn', editor: new Ext.form.field.Checkbox({}), renderer: function(val, meta, rec) {
				    return getStringFromBoolean(val);
				}
				},
                { text: 'Activo?', dataIndex: 'IsActive', editor: new Ext.form.field.Checkbox({}), renderer: function(val, meta, rec) {
                    return getStringFromBoolean(val);
                }
                }
        ],
            plugins: [MasterRowEditorPerson],
            tbar: [
            {
                text: 'Adicionar',
                iconCls: 'add',
                handler: function() {
                    MasterRowEditorPerson.cancelEdit();
                    MasterGridPerson.getStore().insert(0, new Functionary());
                    MasterRowEditorPerson.startEdit(MasterGridPerson.getStore().getAt(0), 0);
                }
            }, '-',
            {
                text: 'Modificar',
                iconCls: 'modify',
                handler: function() {
                    var records = MasterGridPerson.getSelectionModel().getSelection();
                    MasterRowEditorPerson.cancelEdit();
                    MasterRowEditorPerson.startEdit(records[0], 1);
                }
            }, '-',
            {
                text: 'Eliminar',
                iconCls: 'remove',
                handler: function() {
                    var records = MasterGridPerson.getSelectionModel().getSelection();

                    deleteData(
                        aspPageFunctionary,
                        'Delete',
                        'Id_Functionary',
                        records[0].get('Id_Functionary'),
                        function(data) {
                            loadData(aspPageFunctionary, 'List', "{'start':0,'limit':0}", MasterGridPerson.getStore(), null, null);
                        },
                        null
                    );
                }
            }
        ],
            bbar: new Ext.PagingToolbar({
                pageSize: 20,
                store: MasterStorePerson,
                displayInfo: true,
                displayMsg: 'Registros {0} - {1} de {2}',
                emptyMsg: "No hay registros"
            }),
            renderTo: Ext.getBody()
        });

        //loadData(aspPageFunctionary, 'List', "{'start':0,'limit':0}", MasterGridPerson.getStore(), null, null);
    });

File: TMA.MODEL/Entity/Functionary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TMA.MODEL.Entity
{
    [Serializable]
    public class Functionary
    {
        #region Attributes

        private decimal? id_Functionary;
        private int? id_BiometricReader;
        private int? id_Company;
        private int? id_Headquarter;
        private string name;
        private string lastName;
        private int? id_Dependency;
        private int? birthdayMonth;
        private int? birthdayDay;
        private string address;
        private int? id_City;
        private int? postalZone;
        private string airSection;
        private int? telephoneNumber;
        private float? celphoneNumber;
        private string beeperNumber;
        private string beeperCode;
        private int? faxNumber;
        private string email;
        private string password;
        private int? id_Charge;
        private bool byPayroll;
        private bool byHour;
        private bool integralSalary;
        private int? contractClass;
        private bool isActive;
        private int? id_CostCenter;
        private string turn;
        private int? extensionNumber;
        private string bossName;
        private int? bossExtensionNumber;
        private string observations;
        private int? indexReader;
        private int? id_Between;
        private byte[] fingerPrint;
        private byte[] photo;
        private string id_UserCreateRegistration;
        private DateTime? dateCreateRegistration;
        private string id_UserModifyRegistration;
        private DateTime? dateModifyRegistration;

        #endregion Attributes

        #region Properties

        public virtual decimal? Id_Functionary
        {
            get { return id_Functionary; }
            set { id_Functionary = value; }
        }

        public virtual int? Id_BiometricReader
        {
            get { return id_BiometricReader; }
            set { id_BiometricReader = value; }
        }

        public virtual int? Id_Company
        {
            get { return id_Company; }
            set { id_Company = value; }
        }

        public virtual int? Id_Headquarter
        {
            get { return id_Headquarter; }
            set { id_Headquarter = value; }
        }

        public virtual string Name
        {
            get { return name; }
            set { name = value; }
        }

        public virtual string LastName
        {
            get { return lastName; }
            set { lastName = value; }
        }

        public virtual int? Id_Dependency
        {
            get { return id_Dependency; }
            set { id_Dependency = value; }
        }

        public virtual int? BirthdayMonth
        {
            get { return birthdayMonth; }
            set { birthdayMonth = value; }
        }

        public virtual int? BirthdayDay
        {
            get { return birthdayDay; }
            set { birthdayDay = value; }
        }

        public virtual string Address
        {
            get { return address; }
            set { address = value; }
        }

        public virtual int? Id_City
        {
            get { return id_City; }
            set { id_City = value; }
        }

        public virtual int? PostalZone
        {
            get { return postalZone; }
            set { postalZone = value; }
        }

        public virtual string AirSection
        {
            get { return airSection; }
            set { airSection = value; }
        }

        public virtual int? TelephoneNumber
        {
            get { return telephoneNumber; }
            set { telephoneNumber = value; }
        }

        public virtual float? CelphoneNumber
        {
            get { return celphoneNumber; }
            set { celphoneNumber = value; }
        }

        public virtual string BeeperNumber
        {
            get { return beeperNumber; }
            set { beeperNumber = value; }
        }

        public virtual string BeeperCode
        {
            get { return beeperCode; }
            set { beeperCode = value; }
        }

        public virtual int? FaxNumber
        {
            get { return faxNumber; }
            set { faxNumber = value; }
        }

        public virtual string Email
        {
            get { return email; }
            set { email = value; }
        }

        public virtual string Password
        {
            get { return password; }
            set { password = value; }
        }

        public virtual int? Id_Charge
        {
            get { return id_Charge; }
            set { id_Charge = value; }
        }

        public virtual bool ByPayroll
        {
            get { return byPayroll; }
            set { byPayroll = value; }
        }

        public virtual bool ByHour
        {
            get { return byHour; }
            set { byHour = value; }
        }

        public virtual bool IntegralSalary
        {
            get { return integralSalary; }
            set { integralSalary = value; }
        }

        public virtual int? ContractClass
        {
            get { return contractClass; }
            set { contractClass = value; }
        }

        public virtual bool IsActive
        {
            get { return isActive; }
            set { isActive = value; }
        }

        public virtual int? Id_CostCenter
        {
            get { return id_CostCenter; }
            set { id_CostCenter = value; }
        }

        public virtual string Turn
        {
            get { return turn; }
            set { turn = value; }
        }

        public virtual int? ExtensionNumber
        {
            get { return extensionNumber; }
            set { extensionNumber = value; }
        }

        public virtual string BossName
        {
            get { return bossName; }
            set { bossName = value; }
        }

        public virtual int? BossExtensionNumber
        {
            get { return bossExtensionNumber; }
            set { bossExtensionNumber = value; }
        }

        public virtual string Observations
        {
            get { return observations; }
            set { observations = value; }
        }

        public virtual int? IndexReader
        {
            get { return indexReader; }
            set { indexReader = value; }
        }

        public virtual int? Id_Between
        {
            get { return id_Between; }
            set { id_Between = value; }
        }

        public virtual byte[] FingerPrint
        {
            get { return fingerPrint; }
            set { fingerPrint = value; }
        }

        public virtual byte[] Photo
        {
            get { return photo; }
            set { photo = value; }
        }

        public virtual string Id_UserCreateRegistration
        {
            get { return id_UserCreateRegistration; }
            set { id_UserCreateRegistration = value; }
        }

        public virtual DateTime? DateCreateRegistration
        {
            get { return dateCreateRegistration; }
            set { dateCreateRegistration = value; }
        }

        public virtual string Id_UserModifyRegistration
        {
            get { return id_UserModifyRegistration; }
            set { id_UserModifyRegistration = value; }
        }

        public virtual DateTime? DateModifyRegistration
        {
            get { return dateModifyRegistration; }
            set { dateModifyRegistration = value; }
        }

        #endregion Properties
    }
}
